Sources/MapboxNavigation/LaneView.swift

```diff
@@ -98,127 +236,49 @@ open class LaneView: UIView {
-            } else if indications.isSuperset(of: [.uTurn]) {
-                let flip = !(drivingSide == .left)
-                LanesStyleKit.drawLaneUturn(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: flip)
-            } else if indications.isEmpty && isValid {
-                // If the lane indication is `none` and the maneuver modifier has a turn in it,
-                // show the turn in the lane image.
-                if maneuverDirection == .sharpRight || maneuverDirection == .right || maneuverDirection == .slightRight {
-                    if maneuverDirection == .slightRight {
-                        LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                    } else {
-                        LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                    }
-                } else if maneuverDirection == .sharpLeft || maneuverDirection == .left || maneuverDirection == .slightLeft {
-                    if maneuverDirection == .slightLeft {
-                        LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: true)
-                    } else {
-                        LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor, flipHorizontally: true)
-                    }
-                } else {
-                    LanesStyleKit.drawLaneStraight(frame: bounds, resizing: resizing, primaryColor: appropriatePrimaryColor)
-                }
+        let rankedIndications = indications.ranked(favoring: maneuverDirection)
+        guard let laneConfiguration = LaneConfiguration(rankedIndications: rankedIndications, drivingSide: drivingSide) else {
+            return
+        }
+        
+        switch laneConfiguration {
+        case .straightOnly:
+            LanesStyleKit.drawLaneStraight(frame: bounds, resizing: resizing, primaryColor: appropriateColor)
+        case .slightTurnOnly(side: let side):
+            LanesStyleKit.drawLaneSlightRight(frame: bounds, resizing: resizing, primaryColor: appropriateColor,
+                                              flipHorizontally: side == .left)
+        case .turnOnly(side: let side):
+            LanesStyleKit.drawLaneRight(frame: bounds, resizing: resizing, primaryColor: appropriateColor,
+                                        flipHorizontally: side == .left)
+        case .uTurnOnly(side: let side):
+            LanesStyleKit.drawLaneUturn(frame: bounds, resizing: resizing, primaryColor: appropriateColor,
+                                        flipHorizontally: side == .left)
+        case .straightOrTurn(side: let side, straight: let straight, turn: let turn):
+            if isValid && straight && !turn {
```

Comment:
In some cases, we need to distinguish between the straight and turn parts of a drawn lane use diagram. The ranked primary and secondary indications map directly to whether the straight arrow or turn arrow should be highlighted.